app/assets/js/generator.js

```diff
@@ -1,76 +1,113 @@
+    }
+    return item[key];
+  }
+
+  function getUrl(item) {
+    if (item.homepage) {
+      return item.homepage;
+    }
+    if (item.githubRepo) {
+      return 'https://github.com/' + item.githubRepo.user + '/' + item.githubRepo.project;
+    }
+    return 'https://npmjs.com/package/' + item.name;
+  }
+
+  function getDownloadClass(downloads) {
+    if (downloads < 1000) {
+      return null;
+    }
+    if (downloads < 10000) {
+      return 'generator--downloads-t1';
+    }
+    if (downloads < 50000) {
+      return 'generator--downloads-t2';
+    }
+    if (downloads < 100000) {
+      return 'generator--downloads-t3';
+    }
+    return 'generator--downloads-t4';
+  }
+
 
   $(function() {
+
     var instructions = $('#instructions');
     $('#instructions-toggle').on('click', function (e) {
       e.preventDefault();
       instructions.toggle('fast');
     });
 
-    var tpl = doT.template($('#plugins-all-template').text());
-    var pluginsAll = $('#plugins-all');
-    var lastSha = win.Cookies.get('sha');
+    var search = instantsearch({
+      appId: 'OFCNCOG2CU',
+      apiKey: '14ddf54fd4f3435c1cd4038395a0cf10',
+      indexName: 'npm-search',
+      searchParameters: {
+        filters: 'deprecated:false AND keywords:yeoman-generator'
+      }
+    });
 
-    function insertErrorMsg() {
-      pluginsAll.html('There was a problem fetching the generator list. Please try again later.<br />Or you can use <a href="https://npms.io/search?q=keywords%3Ayeoman-generator">npms.io</a>.');
-    }
+    search.addWidget(
+      instantsearch.widgets.searchBox({
+        container: '#searchbox',
+        poweredBy: true,
+        placeholder: 'Search into all yeoman generators'
+      })
+    )
 
-    function fetchAndDisplayListGenerators(sha) {
-      $.getJSON('https://cdn.rawgit.com/yeoman/yeoman-generator-list/' + sha + '/cache.json')
-      .done(function (plugins) {
-        if (Array.isArray(plugins)) {
-          pluginsAll.html(tpl({
-            modules: plugins.sort(function (a, b) {
-              return a.stars === b.stars ? 0 : a.stars < b.stars ? 1 : -1;
-            })
-          }));
+    search.addWidget(
+      instantsearch.widgets.stats({
+        container: '#stats'
+      })
+    );
 
-          var list = new win.List('plugins-all', {
-            valueNames: [
-              'name',
-              'owner',
-              'stars',
-              'updated',
-              'downloads',
-              'description'
-            ]
-          });
+    search.addWidget(
+      instantsearch.widgets.hits({
+        container: '#results',
+        templates: {
+          empty: 'No matching generator found. Try something else.',
+          item: function(item) {
+            var name = highlight(item, 'name').replace('generator-', '')
+            var url = getUrl(item);
+            var description = highlight(item, 'description');
+            var authorName = item.owner.name;
```

Comment:
`var authorName = item.owner.name === "yeoman" ? "The Yeoman Team" : item.owner.name;`

that would do what you suggested @mischah, could even add a custom class or whatever if needed